packages/logic/src/services/ValidationService/Validators/CompareAgePnrValue.ts

```diff
@@ -0,0 +1,29 @@
+import { FDate } from "@fkui/date";
+import { type Validator } from "../Validator";
+import { parseDate, parsePersonnummer } from "../../../converters";
+
+export const compareAgePnrValue: Validator = {
+    name: "compareAgePnrValue",
+    validation(value, _element, config) {
+        let valuePnr = parsePersonnummer(String(value));
+        if (!valuePnr) {
+            return true;
+        }
+        let otherFieldPnr = parsePersonnummer(String(config.otherField));
+        if (!otherFieldPnr) {
+            return true;
+        }
+        valuePnr = parseDate(valuePnr.slice(0, 8));
```

Comment:
Alternativ, mer lättläst? Annars ignorera.

```suggestion
        function toDate(value?: string): string | undefined {
                const datePart = parsePersonnummer(value)?.slice(0, 8);
                if (!datePart) {
                        return undefined;
                }
                
                return FDate.fromIso(parseDate(datePart));          
        }
        
        validation(value, _element, config) {
                const valueAsDate = toDate(value);
                const otherAsDate = toDate(otherFieldPnr);

                if (!valueAsDate || !otherAsDate) {
                        return true;
                }
                
                const pnrValueShouldBeOlder = config.pnrValueShouldBeOlder ? -1: 1;

                return !(FDate.compare(otherAsDate, valueAsDate) === pnrValueShouldBeOlder);
        }            
```

(returnen blev inget vidare)

Reply:
Håller med, jag hade nog flyttat ut funktionen till global scope men annars blir det väldigt snyggt med att ta emot `undefined` (kanske borde ta `unknown` dock) så blir det mindre repetetivt.